strawberryfields/utils/post_processing.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module provides functions for post-processing samples.
+"""
+import numpy as np
+
+
+def number_expectation_pnr(photon_number_samples, modes=None):
+    """The expectation value of the number operator from PNR samples.
+
+    **Example:**
+
+    .. code-block:: python
+
+        samples = np.array([[2, 0],
+                            [2, 2],
+                            [2, 0],
+                            [0, 0]])
+
+    Getting the expectation value for these PNR samples:
+
+    >>> number_expectation_pnr(samples)
+    1.0
```

Comment:
If possible, I would rewrite this to use a single `>>>` block if possible. Maybe like this?

```pycon
>>> samples = np.array([[2, 0], [2, 2], [2, 0], [0, 0]])
>>> number_expectation_pnr(samples)
```

Reply:
Thanks! Is this mostly such that no extra space is taken up in the docs?

Reply:
More so that if we decide to use `doctest` in the future :)